routers/api/v1/repo/board.go

```diff
@@ -0,0 +1,231 @@
+	// ---
+	// summary: Get project board
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the board
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ProjectBoard"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	board, err := project_model.GetBoard(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		if project_model.IsErrProjectBoardNotExist(err) {
+			ctx.Error(http.StatusNotFound, "GetProjectBoard", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetProjectBoard", err)
+		}
+		return
+	}
+
+	if board.Project.RepoID != ctx.Repo.Repository.ID {
+		ctx.Error(http.StatusUnauthorized, "GetProjectBoard", "board doesn't belong to the repository")
+		return
+	}
+
+	ctx.JSON(http.StatusOK, convert.ToAPIProjectBoard(board))
+}
+
+func UpdateProjectBoard(ctx *context.APIContext) {
+	// swagger:operation PATCH /projects/boards/{id} board boardUpdateProjectBoard
+	// ---
+	// summary: Update project board
+	// produces:
+	// - application/json
+	// consumes:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project board
+	//   type: string
+	//   required: true
+	// - name: board
+	//   in: body
+	//   required: true
+	//   schema: { "$ref": "#/definitions/UpdateProjectBoardPayload" }
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ProjectBoard"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	form := web.GetForm(ctx).(*api.UpdateProjectBoardPayload)
+
+	board, err := project_model.GetBoard(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "GetProjectBoard", err)
+		return
+	}
+
+	if board.Title != form.Title {
+		board.Title = form.Title
+	}
+	if board.Color != form.Color {
+		board.Color = form.Color
+	}
+
+	if err = project_model.UpdateBoard(ctx, board); err != nil {
+		ctx.Error(http.StatusInternalServerError, "UpdateProjectBoard", err)
+		return
+	}
+
+	board, err = project_model.GetBoard(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "GetProjectBoard", err)
+		return
+	}
+
+	ctx.JSON(http.StatusOK, convert.ToAPIProjectBoard(board))
+}
+
+func DeleteProjectBoard(ctx *context.APIContext) {
+	// swagger:operation DELETE /projects/boards/{id} board boardDeleteProjectBoard
+	// ---
+	// summary: Delete project board
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project board
+	//   type: string
+	//   required: true
+	// responses:
+	//   "204":
+	//     "description": "Project board deleted"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	if err := project_model.DeleteBoardByID(ctx.ParamsInt64(":id")); err != nil {
```

Comment:
You should also check the board's repository id is equal to context repository id.

Reply:
I don't think we have access to the repository on these routes since these routes are not repo or project dependent

Reply:
@lunny There is the permission check above, to confirm user can write to projects, and it is based on the repo linked to the project.